core/crypto/kem/hybrid/hybrid.go

```diff
@@ -0,0 +1,329 @@
+		first:  first,
+		second: second,
+	}
+}
+
+func (sch *Scheme) Name() string { return sch.name }
+func (sch *Scheme) PublicKeySize() int {
+	return sch.first.PublicKeySize() + sch.second.PublicKeySize()
+}
+
+func (sch *Scheme) PrivateKeySize() int {
+	return sch.first.PrivateKeySize() + sch.second.PrivateKeySize()
+}
+
+func (sch *Scheme) SeedSize() int {
+	return SeedSize
+}
+
+func (sch *Scheme) SharedKeySize() int {
+	return sch.first.SharedKeySize() + sch.second.SharedKeySize()
+}
+
+func (sch *Scheme) CiphertextSize() int {
+	return sch.first.CiphertextSize() + sch.second.CiphertextSize()
+}
+
+func (sch *Scheme) EncapsulationSeedSize() int {
+	return SeedSize
+}
+
+func (sk *PrivateKey) Scheme() kem.Scheme { return sk.scheme }
+func (pk *PublicKey) Scheme() kem.Scheme  { return pk.scheme }
+
+func (sk *PrivateKey) MarshalBinary() ([]byte, error) {
+	if sk.first == nil || sk.second == nil {
+		return nil, ErrUninitialized
+	}
+	first, err := sk.first.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	second, err := sk.second.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	return append(first, second...), nil
+}
+
+func (sk *PrivateKey) Equal(other kem.PrivateKey) bool {
+	oth, ok := other.(*PrivateKey)
+	if !ok {
+		return false
+	}
+	if sk.first == nil && sk.second == nil && oth.first == nil && oth.second == nil {
+		return true
+	}
+	if sk.first == nil || sk.second == nil || oth.first == nil || oth.second == nil {
+		return false
+	}
+	return sk.first.Equal(oth.first) && sk.second.Equal(oth.second)
+}
+
+func (sk *PrivateKey) Public() kem.PublicKey {
+	return &PublicKey{sk.scheme, sk.first.Public(), sk.second.Public()}
+}
+
+func (pk *PublicKey) Equal(other kem.PublicKey) bool {
+	oth, ok := other.(*PublicKey)
+	if !ok {
+		return false
+	}
+	if pk.first == nil && pk.second == nil && oth.first == nil && oth.second == nil {
+		return true
+	}
+	if pk.first == nil || pk.second == nil || oth.first == nil || oth.second == nil {
+		return false
+	}
+	return pk.first.Equal(oth.first) && pk.second.Equal(oth.second)
+}
+
+func (pk *PublicKey) MarshalBinary() ([]byte, error) {
+	if pk.first == nil || pk.second == nil {
+		return nil, ErrUninitialized
+	}
+	first, err := pk.first.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	second, err := pk.second.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	return append(first, second...), nil
+}
+
+func (sch *Scheme) GenerateKeyPair() (kem.PublicKey, kem.PrivateKey, error) {
+	pk1, sk1, err := sch.first.GenerateKeyPair()
+	if err != nil {
+		return nil, nil, err
+	}
+	pk2, sk2, err := sch.second.GenerateKeyPair()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &PublicKey{sch, pk1, pk2}, &PrivateKey{sch, sk1, sk2}, nil
+}
+
+func (sch *Scheme) DeriveKeyPair(seed []byte) (kem.PublicKey, kem.PrivateKey) {
+	if len(seed) != sch.SeedSize() {
+		panic(kem.ErrSeedSize)
+	}
+
+	h, err := blake2b.NewXOF(uint32(sch.first.SeedSize()+sch.second.SeedSize()), nil)
```

Comment:
I think it would be nice to just require `len(seed) == sch.first.SeedSize()+sch.second.SeedSize()` and then passing slices on to `sch.XXX.DeriveKeyPair(slice_XXXX)` here, instead of the force `blake2b.NewXOF`  step?

Reply:
fixed!